packages/compass-data-modeling/src/utils/schema-traversal.tsx

```diff
@@ -165,3 +165,118 @@ export const getFieldFromSchema = ({
     parentFieldPath: [...parentFieldPath, nextInPath],
   });
 };
+
+type MutationParameters = {
+  update: 'removeField' | 'renameField';
+  newFieldName?: string;
+};
+
+const getMutatedSchema = ({
```

Comment:
Nit: mutation has a certain meaning in JS usually referring to the original object getting modified directly, you're not mutating it, you're just returning a new one updated, so I'd suggest to rename to represent that